### ISSUE TITLE
Mutation of enum

### ISSUE THREAD

**Issue by tjchambers** [mbj/mutant#371]
I have at least 2 models in Rails that have enum usage. Here is an example:

```
  enum seq: [:bad, :goal, :unintended, :barrier, :assumption]
```

When I attempt to mutate these models I get output on the console (no output in the log):

```
/Users/tj/.rvm/gems/ruby-2.2.0@global/gems/activerecord-4.2.1/lib/active_record/enum.rb:188:in `detect_enum_conflict!': You tried to define an enum named "seq" on the model "GubaCategory", but this will generate a instance method "seq=", which is already defined by another enum. (ArgumentError)
	from /Users/tj/.rvm/gems/ruby-2.2.0@global/gems/activerecord-4.2.1/lib/active_record/enum.rb:95:in `block (2 levels) in enum'
	from /Users/tj/.rvm/gems/ruby-2.2.0@global/gems/activerecord-4.2.1/lib/active_record/enum.rb:93:in `module_eval'
	from /Users/tj/.rvm/gems/ruby-2.2.0@global/gems/activerecord-4.2.1/lib/active_record/enum.rb:93:in `block in enum'
	from /Users/tj/.rvm/gems/ruby-2.2.0@global/gems/activerecord-4.2.1/lib/active_record/enum.rb:84:in `each'
	from /Users/tj/.rvm/gems/ruby-2.2.0@global/gems/activerecord-4.2.1/lib/active_record/enum.rb:84:in `enum'
	from /Users/tj/projects/AOT/aotv2/app/models/align/guba_category.rb:41:in `<class:GubaCategory>'
	from /Users/tj/projects/AOT/aotv2/app/models/align/guba_category.rb:7:in `<top (required)>'
	from /Users/tj/.rvm/gems/ruby-2.2.0@global/bundler/gems/mutant-8e297809610a/lib/mutant/env/bootstrap.rb:105:in `require'
	from /Users/tj/.rvm/gems/ruby-2.2.0@global/bundler/gems/mutant-8e297809610a/lib/mutant/env/bootstrap.rb:105:in `each'
	from /Users/tj/.rvm/gems/ruby-2.2.0@global/bundler/gems/mutant-8e297809610a/lib/mutant/env/bootstrap.rb:105:in `infect'
	from /Users/tj/.rvm/gems/ruby-2.2.0@global/bundler/gems/mutant-8e297809610a/lib/mutant/env/bootstrap.rb:37:in `initialize'
	from /Users/tj/.rvm/gems/ruby-2.2.0@global/gems/adamantium-0.2.0/lib/adamantium/class_methods.rb:17:in `new'
	from /Users/tj/.rvm/gems/ruby-2.2.0@global/gems/adamantium-0.2.0/lib/adamantium/class_methods.rb:17:in `new'
	from /Users/tj/.rvm/gems/ruby-2.2.0@global/bundler/gems/mutant-8e297809610a/lib/mutant/env/bootstrap.rb:26:in `new'
	from /Users/tj/.rvm/gems/ruby-2.2.0@global/gems/procto-0.0.2/lib/procto.rb:62:in `block in initialize'
	from /Users/tj/.rvm/gems/ruby-2.2.0@global/bundler/gems/mutant-8e297809610a/lib/mutant/cli.rb:23:in `run'
	from /Users/tj/.rvm/gems/ruby-2.2.0@global/bundler/gems/mutant-8e297809610a/bin/mutant:35:in `<top (required)>'
	from /Users/tj/.rvm/gems/ruby-2.2.0@global/bin/mutant:23:in `load'
	from /Users/tj/.rvm/gems/ruby-2.2.0@global/bin/mutant:23:in `<main>'
	from /Users/tj/.rvm/gems/ruby-2.2.0@global/bin/ruby_executable_hooks:15:in `eval'
	from /Users/tj/.rvm/gems/ruby-2.2.0@global/bin/ruby_executable_hooks:15:in `<main>'
```

My other model is the same output with reference to it's enum and not mutant logged output whatsoever. As this is a new feature I presume it is not covered by tests yet.

**Comment by mbj** [comment]
@tjchambers Seems that some hook inside AR triggers that prevents mutant from inserting its mutation. I'd apprechiate a reproducing repository so I can see if there is an easy fix. But I doubt such a fix can land in master as it might be AR/AS specific.

**Comment by tjchambers** [comment]
You got it - will have a repo for that today too. Understand. Would be nice to workaround enums if they are difficult to support.

**Comment by mbj** [comment]
> Would be nice to workaround enums if they are difficult to support.

There is no `enum` concept in ruby itself. Its an abstraction AR adds. Its very likely the implementation of the enum abstraction in AR that relies on some kind of hooks system is the issue here.

**Comment by tjchambers** [comment]
My inference is that enums do not "look" like anything identifiable and therefore would be hard to distinguish.

**Comment by tjchambers** [comment]
BTW - My inference is that something changed recently that surfaced this. I have had enums for a number of months, and was able to previously mutate one of these modules. The fact that this does not mutate today is a bit unexpected. Again I don't have specifics (i.e. mutant version number, date when I last mutated this code, etc.). I can only say it seems a bit odd that this has surfaced. Not discounting that I may have upgraded something that triggered it as well.

**Comment by mbj** [comment]
@tjchambers Can also be an implementation change in AR.

**Comment by tjchambers** [comment]
During my struggles to get mutant to fail in a small test reproduction of this, I discovered that reloading the class is known to be problematic with enum's as currently implemented in rails. So upon further investigation I changed my enums to read:

```
enum seq: [:bad, :goal, :unintended, :barrier, :assumption] unless instance_methods.include? :seq
```

which inhibits attempts to redefine them. After rerunning my same rake task over all of my prior classes containing enums, I encountered NO errors. So I am closing this as this is a simple and suitable workaround. 

I do not know if my rake task or mutant is the trigger for the class reload, but that is irrelevant because the workaround is without negative side effects.
